packages/fxa-settings/src/pages/ResetPassword/ResetPasswordRecoveryPhone/index.test.tsx

```diff
@@ -0,0 +1,137 @@
+    verifyCode: mockVerifyCodeError,
+    resendCode: mockResendCodeError,
+  };
+
+  beforeEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('renders as expected', async () => {
+    renderWithLocalizationProvider(<ResetPasswordRecoveryPhone {...defaultProps} />);
+
+    expect(screen.getByRole('button', { name: 'Back' })).toBeInTheDocument();
+    expect(
+      screen.getByRole('heading', { name: 'Reset your password' })
+    ).toBeInTheDocument();
+    expect(
+      screen.getByRole('heading', { name: 'Enter recovery code' })
+    ).toBeInTheDocument();
+    expect(
+      screen.getByRole('textbox', { name: 'Enter 6-digit code' })
+    ).toBeInTheDocument();
+    expect(screen.getByText('Confirm')).toBeInTheDocument();
+    expect(
+      screen.getByRole('button', { name: 'Resend code' })
+    ).toBeInTheDocument();
+    expect(
+      screen.getByRole('link', {
+        name: 'Are you locked out? Opens in new window',
+      })
+    ).toBeInTheDocument();
+  });
+
+  it('has expected glean click events', async () => {
+    const user = userEvent.setup();
+    renderWithLocalizationProvider(<ResetPasswordRecoveryPhone {...defaultProps} />);
+
+    await waitFor(() => user.type(screen.getByRole('textbox'), '123456'));
+
+    expect(screen.getByRole('button', { name: /Confirm/i })).toHaveAttribute(
+      'data-glean-id',
+      'reset_password_backup_phone_submit'
+    );
+
+    const resendButton = screen.getByRole('button', { name: /Resend code/i });
+    expect(resendButton).toHaveAttribute(
+      'data-glean-id',
+      'reset_password_backup_phone_resend'
+    );
+    const lockedOutLink = screen.getByRole('link', {
+      name: /Are you locked out\?/i,
+    });
+    expect(lockedOutLink).toHaveAttribute(
+      'data-glean-id',
+      'reset_password_backup_phone_locked_out_link'
+    );
+  });
+
+  it('submits with valid code', async () => {
+    renderWithLocalizationProvider(<ResetPasswordRecoveryPhone {...defaultProps} />);
+
+    const input = screen.getByRole('textbox');
+
+    await waitFor(() => userEvent.type(input, '123456'));
+    userEvent.click(screen.getByRole('button', { name: 'Confirm' }));
+    await waitFor(() => {
+      expect(mockVerifyCode).toHaveBeenCalledWith('123456');
+    });
+  });
+
+  it('handles invalid code', async () => {
+    renderWithLocalizationProvider(<ResetPasswordRecoveryPhone {...propsWithError} />);
+
+    const input = screen.getByRole('textbox');
+
+    await waitFor(() => userEvent.type(input, '123456'));
+    userEvent.click(screen.getByRole('button', { name: 'Confirm' }));
+    await waitFor(() => {
+      expect(mockVerifyCodeError).toHaveBeenCalledWith('123456');
+    });
+    expect(screen.getByRole('alert')).toHaveTextContent(
+      /The code is invalid or expired./
+    );
+    expect(
+      screen.getByRole('link', {
+        name: 'Use backup authentication codes instead?',
+      })
+    ).toHaveAttribute('href', '/reset_password_recovery_code');
```

Comment:
The actual path is `/confirm_backup_code_reset_password` (should be updated throughout 😄) (Though I do like the idea of having all reset password paths start with `reset_password`

Reply:
Also noticed that we should make sure that all the paths for 2FA in password reset are listed in content-server, otherwise reloading the urls leads to a `Page not found` error.